akka-sample-sharding-scala/killrweather/src/main/scala/sample/killrweather/WeatherHttpServer.scala

```diff
@@ -0,0 +1,52 @@
+package sample.killrweather
+
+import scala.util.{Failure, Success}
+import scala.concurrent.duration._
+
+import akka.actor.typed.ActorSystem
+import akka.actor.CoordinatedShutdown
+import akka.{Done, actor => classic}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.server.Route
+
+private[killrweather] object WeatherHttpServer {
+
+  /**
+   * Logic to bind the given routes to a HTTP port and add some logging around it
+   */
+  def start(routes: Route, port: Int, system: ActorSystem[_]): Unit = {
+    import akka.actor.typed.scaladsl.adapter._
+    implicit val classicSystem: classic.ActorSystem = system.toClassic
+    val shutdown = CoordinatedShutdown(classicSystem)
+
+    import system.executionContext
+
+    Http().bindAndHandle(routes, "localhost", port).onComplete {
+      case Success(binding) =>
+        val address = binding.localAddress
+        system.log.info(
+          "WeatherServer online at http://{}:{}/",
+          address.getHostString,
+          address.getPort
+        )
+
+        shutdown.addTask(
+          CoordinatedShutdown.PhaseServiceRequestsDone,
+          "http-graceful-terminate"
+        ) { () =>
+          binding.terminate(10.seconds).map { _ =>
+            system.log.info(
+              "WeatherServer http://{}:{}/ graceful shutdown completed",
+              address.getHostString,
+              address.getPort
+            )
+            Done
+          }
+        }
+      case Failure(ex) =>
+        system.log.error("Failed to bind HTTP endpoint, terminating system", ex)
+        system.terminate()
+    }
+  }
+
+}
```

Comment:
Made this a single object call since the object only was created to call that method anyway and didn't have any lifecycle or other methods operating on the class fields